### PR TITLE
[Typing][C-25,C-26] Add type annotations for `python/paddle/distributed/communication/stream/{all_reduce,all_to_all}.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description
类型标注：

- python/paddle/distributed/communication/stream/all_reduce.py
- python/paddle/distributed/communication/stream/all_to_all.py

### Related links
- https://github.com/PaddlePaddle/Paddle/issues/65008
@megemini

### CODE REVIEW

File: python/paddle/distributed/communication/stream/all_to_all.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Sequence
```
Comment (megemini) [inline python/paddle/distributed/communication/stream/all_to_all.py]: 从 `collections.abc` 导入 `Sequence` ～ 参考 https://github.com/PaddlePaddle/Paddle/issues/66936

**Review by megemini — COMMENTED** [review]
示例里面加个 `ignore` 吧，CI 提示有两个地方

``` python
>>> task.wait() # type: ignore[union-attr]
```

File: python/paddle/distributed/communication/stream/all_to_all.py (hunk 1 above)
Comment (enkilee) [reply python/paddle/distributed/communication/stream/all_to_all.py]: 收到

**Review by megemini — APPROVED** [review]
LGTM ~

**Review by luotao1 — APPROVED** [review]

**Pull request merged**